### ISSUE TITLE
As a GM, I want to be notified that I'm going to end the game when refreshing my tab

### ISSUE THREAD

**Issue by Stoob** [fariapp/fari#178]
## 🚩 Describe the bug
When you click the Refresh or Reload button in your browser, the game is ended and the game link no longer works

## 🔢 Steps to reproduce

1. Create a game
2. Load a scene 
3. Add characters
4. Click the Refresh or Reload button in your browser

## 🟢 Expected behavior

Game state stays active when refresh

## 🔴 Actual behavior

Game link stops working, game ends.  Scene can be reloaded (if saved) but characters must be re-added to game.  PCs see game 'does not exist' must rejoin at a new link

## 🖥 Desktop or Smartphone

- OS: any
- Browser: any
- Device: any

## 📄 Additional context

Maybe this is a limitation of the technology in use, but it just seems odd to me that clicking a single button ends the game entirely and forces re-creation of a new game.  I learned this the hard way.  ;-) Is there any way to preserve the game state or keep a 'game link' active?  Such as registering for a user account, or having a persistent link?  Interestingly as a player I see that I must rejoin the game if I refresh my browser.

**Comment by RPDeshaies** [comment]
Yeah sadly that's because the GM is the "host" of the game and if the GM leaves, everyone's connection is terminated.

There's no work around for this apart for you to "save" the scene before leaving.

I realized I forgot to add a warning popup to try to stop situations like this from hapenning. I converted the bug to implement this instead because there's no way to "keep the conneciton alive" if the GM leaves
